src/components/quantity-input/quantity-input.tsx

```diff
@@ -4,60 +4,56 @@ import styles from './quantity-input.module.scss';
 import { useRef } from 'react';
 
 type QuantityInputProps = {
-    value: number | null;
-    onChange: (value: number | null) => void;
+    value: number;
```

Comment:
Let's do it like this:

1. On input change, strip non-digits (e.g. typing -5 or 5.5 should not be allowed). If the value is an empty string or zero, don't emit an event.
2. On blur, revert to the value from props.
3. Increment and decrement should round the number.

Reply:
https://stackblitz.com/edit/vitejs-vite-rxpkne?file=src%2FApp.tsx